### PR TITLE
Add Qwen3 voice models

### LINKED ISSUE #4135: Add model: Add qwen-voice-embedding
### Model link on Hugging Face

https://huggingface.co/collections/marksverdhei/qwen3-voice-embedding

### Arxiv link

_No response_

### Description of the model

_No response_

### PR DESCRIPTION
**Pull request by deep9539**
Closes: #4135 

- [x] I have filled out the ModelMeta object to the extent possible
- [x] I have ensured that my model can be loaded using
  - [x] `mteb.get_model(model_name, revision)` and
  - [x] `mteb.get_model_meta(model_name, revision)`
- [x] I have tested the implementation works on a representative set of tasks.
- [x] The model is public, i.e., is available either as an API or the weights are publicly available to download
- [N/A] I reproduced results from the original paper (if applicable) on at least one benchmark, and I am including the results in the PR description.

Qwen3-Voice-Embedding-12Hz-1.7B on IEMOCAPEmotion: 0.200719 (Couldn't find any Qwen3-voice related numbers)
This number however looks in the range other embedding models performs for this task based on page 24 (https://arxiv.org/pdf/2602.16008)

### CODE REVIEW

File: mteb/models/model_implementations/qwen3_voice_models.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    loader=Qwen3VoiceEmbeddingWrapper,
+    name="marksverdhei/Qwen3-Voice-Embedding-12Hz-0.6B",
+    languages=["eng-Latn"],
+    open_weights=True,
+    revision="93f9dff7198816748fd8263b03298b351ca36cd8",
+    release_date="2026-02-09",
+    max_tokens=float("inf"),
+    n_parameters=8_854_336,
+    n_embedding_parameters=0,
+    memory_usage_mb=None,
```
Comment (Samoed) [inline mteb/models/model_implementations/qwen3_voice_models.py]: Can you fill this

File: mteb/models/model_implementations/qwen3_voice_models.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        inputs.collate_fn = AudioCollator(target_sampling_rate=self.sampling_rate)
+        all_embeddings = []
+
+        max_samples = int(self.max_audio_length_seconds * self.sampling_rate)
+
+        for batch in tqdm(
+            inputs,
+            disable=not show_progress_bar,
+        ):
+            audio_arrays = [audio["array"][:max_samples] for audio in batch["audio"]]
```
Comment (Samoed) [inline mteb/models/model_implementations/qwen3_voice_models.py]: You can set max_samples in collator
```suggestion
```

File: mteb/models/model_implementations/qwen3_voice_models.py (hunk 1 above)
Comment (deep9539) [reply mteb/models/model_implementations/qwen3_voice_models.py]: Copying from agent:

In MTEB, the n_embedding_parameters attribute represents the number of parameters in the model's vocabulary lookup/token embedding layer (typically a PyTorch nn.Embedding layer). 

**Why it is 0 for Qwen3 Voice Embedding?**
Pure audio encoders and vision encoders (like EcapaTdnnSpeakerEncoder, standard Audio Spectrogram Transformers (AST), and CNNs):
   1. Do not use a text vocabulary or a discrete token tokenizer.
   2. Accept continuous inputs (such as raw log-mel spectrogram features, raw audio waves, or raw pixel patches) directly into convolutional or linear projection layers.
   3. Contain no discrete input lookup table (nn.Embedding layer).
   
 (Resolving, please feel free to reopen)

File: mteb/models/model_implementations/qwen3_voice_models.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    loader=Qwen3VoiceEmbeddingWrapper,
+    name="marksverdhei/Qwen3-Voice-Embedding-12Hz-0.6B",
+    languages=["eng-Latn"],
+    open_weights=True,
+    revision="93f9dff7198816748fd8263b03298b351ca36cd8",
+    release_date="2026-02-09",
+    max_tokens=float("inf"),
+    n_parameters=8_854_336,
+    n_embedding_parameters=0,
+    memory_usage_mb=17,
```
Comment (Samoed) [inline mteb/models/model_implementations/qwen3_voice_models.py:138]: 0.6b model can't use only 17mb of ram

File: mteb/models/model_implementations/qwen3_voice_models.py (hunk 1 above)
Comment (Samoed) [reply mteb/models/model_implementations/qwen3_voice_models.py]: I didn't ask about `n_embedding_parameters`, I ask about only `memory_usage_mb`

File: mteb/models/model_implementations/qwen3_voice_models.py (hunk 3 above)
Comment (deep9539) [reply mteb/models/model_implementations/qwen3_voice_models.py:138]: Yeah, good question! 

The "0.6B" in the repo name refers to the source model it was extracted from, not the size of this model itself.

Here's what's going on: this repo is a standalone ECAPA-TDNN speaker/voice encoder that was extracted from Qwen/Qwen3-TTS-12Hz-0.6B-Base (a 0.6-billion-parameter TTS model). The naming convention Qwen3-Voice-Embedding-12Hz-0.6B is telling you "this was pulled out of the 0.6B TTS model," similar to how the related 1.7B variant was pulled from the 1.7B TTS model.

But the actual encoder itself — the ECAPA-TDNN component that turns a mel-spectrogram into a 1024-dim x-vector speaker embedding — is tiny: the model card lists ~6.3M parameters. That's consistent with a ~17MB footprint (roughly 6.3M params × ~2-4 bytes/param in fp16/fp32, plus overhead ≈ in that ballpark).

So the full Qwen3-TTS-0.6B model includes a text encoder, decoder, and other components totaling ~0.6B parameters, but only the small speaker-encoding sub-module (ECAPA-TDNN) was extracted and published here — hence the mismatch between the "0.6B" in the name and the tiny actual file size.

Comment (deep9539) [reply mteb/models/model_implementations/qwen3_voice_models.py:138]: (Please feel free to reopen this)

**Review by Samoed — APPROVED** [review]
